src/server/pegasus_server_impl.cpp

```diff
@@ -1326,16 +1326,16 @@ ::dsn::error_code pegasus_server_impl::start(int argc, char **argv)
 
     ddebug("%s: start to open rocksDB's rdb(%s)", replica_name(), path.c_str());
 
-    bool need_create_meta_cf = true;
-    // Check meta CF only when db exist.
-    if (db_exist && check_meta_cf(path, &need_create_meta_cf) != ::dsn::ERR_OK) {
-        derror_replica("check meta column family failed");
-        return ::dsn::ERR_LOCAL_APP_FAILURE;
-    }
-    if (need_create_meta_cf) {
-        // If upgrade from an old Pegasus version which has just one column family (the default
-        // column family), or create new db, we have to create a new column family to store meta
-        // data (meta column family).
+    if (db_exist) {
+        // When DB exist, meta CF must be present.
+        bool missing_meta_cf = true;
+        if (check_meta_cf(path, &missing_meta_cf) != ::dsn::ERR_OK) {
+            derror_replica("check meta column family failed");
+            return ::dsn::ERR_LOCAL_APP_FAILURE;
+        }
+        dassert_replica(!missing_meta_cf, "You must upgrade Pegasus server from 2.0");
```

Comment:
“NOTE: Since meta data is still wrote into manifest, so there is no compatibility problems with 2.0.0.”

I understand the pr resolve the `compatibility problems`, so is here still dassert because `compatibility`?

Reply:
This pr just keep compatibility with 2.0.0, not compatible with 1.x, so if upgrade from 1.x, it will dassert false here.